postal/internal/dependency_mappings.go

```diff
@@ -23,19 +24,31 @@ func NewDependencyMappingResolver(bindingResolver BindingResolver) DependencyMap
 }
 
 // FindDependencyMapping looks up if there is a matching dependency mapping
-func (d DependencyMappingResolver) FindDependencyMapping(sha256, platformDir string) (string, error) {
+// If the binding is given in the form of `hash`, assume it is of algorithm `sha256`
+// If the binding is given in the form of `algorithm:hash`, compare it to the full `checksum` input
+func (d DependencyMappingResolver) FindDependencyMapping(checksum, platformDir string) (string, error) {
 	bindings, err := d.bindingResolver.Resolve("dependency-mapping", "", platformDir)
 	if err != nil {
 		return "", fmt.Errorf("failed to resolve 'dependency-mapping' binding: %w", err)
 	}
 
 	for _, binding := range bindings {
-		if uri, ok := binding.Entries[sha256]; ok {
+		hash := cargo.Checksum(checksum).Hash()
```

Comment:
You can move this outside the loop so it doesn't get re-evaluated on each iteration.